migration/migration_test.go

```diff
@@ -0,0 +1,48 @@
+package migration
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConcurrentMigrations(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	var err error
+	if err = configuration.Setup(""); err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < 20; i++ {
+		wg.Add(1)
+		db, err := gorm.Open("postgres",
+			fmt.Sprintf("host=%s port=%d user=%s password=%s DB.name=%s sslmode=%s",
+				configuration.GetPostgresHost(),
+				configuration.GetPostgresPort(),
+				configuration.GetPostgresUser(),
+				configuration.GetPostgresPassword(),
+				configuration.GetPostgresDatabase(),
+				configuration.GetPostgresSSLMode(),
+			))
+		if err != nil {
+			t.Fatal("Cannot connect to DB", err)
+		}
+
+		go func(db *gorm.DB) {
+			defer wg.Done()
+			err = Migrate(db)
```

Comment:
@kwk Our integration suite should verify the migration in general, so I'm not sure we need to verify that as part of this test. The current test just sees it won't fail anyway, which is technically what the integration suite does over and over again. The individual tests would check if the individual tables work as well.

Maybe this test should focus on the migration code. Possibly have an option to define a 'test-version' table and some custom migration scripts to see it can actually do what it's suppose to. Then we can easially add some multi threading and sleeps.